conan_provider.cmake

```diff
@@ -425,6 +425,17 @@ function(detect_host_profile output_file)
     string(APPEND profile "[conf]\n")
     string(APPEND profile "tools.cmake.cmaketoolchain:generator=${CMAKE_GENERATOR}\n")
 
+    # if a custom cmake toolchain was specified use it
+    if(CMAKE_TOOLCHAIN_FILE AND NOT EXISTS ${CMAKE_TOOLCHAIN_FILE})
+      message(FATAL_ERROR "Specified toolchian file ${CMAKE_TOOLCHAIN_FILE} does not exist")
+    endif()
```

Comment:
Is this really necessary? I mean, if the toolchain file is not there when cmake is invoked, cmake will already error, so it will never get to this point in the ``conan_provider.cmake``?